coffee/cse.py

```diff
@@ -411,7 +412,7 @@ def _cost_fact(self, trace, levels, lda, bounds):
                     else:
                         handle = [read]
                     linear_reads.extend(handle)
-                factors = {as_urepr(i): i for i in linear_reads}.values()
+                factors = list(itervalues({as_urepr(i): i for i in linear_reads}))
```

Comment:
you don't need determinism there because the info is just used by the cost model to eventually produce a number